app/src/main/java/org/mozilla/tv/firefox/pocket/PocketRepo.kt

```diff
@@ -0,0 +1,125 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.tv.firefox.pocket
+
+import android.arch.lifecycle.LiveData
+import android.arch.lifecycle.MutableLiveData
+import android.os.SystemClock
+import android.support.annotation.UiThread
+import kotlinx.coroutines.experimental.CancellationException
+import kotlinx.coroutines.experimental.Job
+import kotlinx.coroutines.experimental.delay
+import kotlinx.coroutines.experimental.launch
+import org.mozilla.tv.firefox.utils.BuildConfigDerivables
+import java.util.concurrent.TimeUnit
+import kotlin.math.min
+
+private val CACHE_UPDATE_FREQUENCY_MILLIS = TimeUnit.MINUTES.toMillis(45)
+
+sealed class PocketRepoState {
+    data class LoadComplete(val videos: List<PocketFeedItem>) : PocketRepoState()
+    object Loading : PocketRepoState()
+    object NoKey : PocketRepoState()
+    object FetchFailed : PocketRepoState()
+}
+
+/**
+ * Manages backing state for Pocket data, as well as any logic related to
+ * retrieving or storing that data.
+ */
+open class PocketRepo(
+    private val pocketEndpoint: PocketEndpoint,
+    private val pocketRepoStateMachine: PocketRepoStateMachine,
+    buildConfigDerivables: BuildConfigDerivables
+) {
+
+    private val mutableState = MutableLiveData<PocketRepoState>().apply {
```

Comment:
Curious about the purpose of having both `mutableState` and `state`. Why did you decide to do this?

Reply:
It's a backing field, equivalent to `_pinnedTiles` in your PR.  I'll update the name to make that more clear.